clippy_lints/src/utils/mod.rs

```diff
@@ -911,6 +911,7 @@ pub fn is_ctor_or_promotable_const_function(cx: &LateContext<'_, '_>, expr: &Exp
         if let ExprKind::Path(ref qp) = fun.kind {
             let res = cx.tables.qpath_res(qp, fun.hir_id);
             return match res {
+                def::Res::Def(DefKind::Fn, def_id) => rustc_mir::const_eval::is_const_fn(cx.tcx, def_id),
```

Comment:
There are another way: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/struct.FnHeader.html#structfield.constness

Reply:
No, this is definitely the right way. The constness field by itself is set, even if the function's constness is unstable.